inc/fasta/fasta.hpp

```diff
@@ -34,8 +33,28 @@ namespace ebi
       class IFasta
       {
       public:
+        /**
+         * Extract n base pairs from the contig starting from an offset
+         * @param contig - the name of the contig to extract the sequence from
+         * @param pos - the starting offset position
+         * @param length - the number of base pairs to be extracted
+         * @return the sequence string, empty if nothing can be extracted.
+         */
         virtual std::string sequence(const std::string& contig, const size_t pos, const size_t length) = 0;
-        virtual size_t count(const std::string& contig) const = 0;
+
+        /**
+         * Get the number of times a contig appears in a FASTA.
+         * @param contig - the name of the contig
+         * @return the number of times a contig appears in a FASTA. 0 if the contig is not found.
+         */
+        virtual size_t count(const std::string &contig) const = 0;
+
+        /**
+         * Get the length of the sequence for a contig
+         * @param contig - the name of the contig
+         * @return the length of the sequence for the contig. 0 if the contig is not found.
+         */
+        virtual size_t len(const std::string &contig) const = 0;
```

Comment:
unnecessary abbreviation, in my opinion. Actually, if you think it's useful to say in the documentation that it's the *sequence* length, I would call the method `sequence_length`.